iceoryx_posh/source/popo/wait_set.cpp

```diff
@@ -107,23 +103,33 @@ WaitSet::ConditionVector WaitSet::createVectorWithFullfilledConditions() noexcep
     {
         if (currentCondition->hasTriggered())
         {
-            if (!conditions.push_back(currentCondition))
-            {
-                errorHandler(Error::kPOPO__WAITSET_CONDITION_VECTOR_OVERFLOW, nullptr, ErrorLevel::FATAL);
-            }
+            // We do not need to verify if push_back was successful since
+            // m_conditionVector and conditions are having the same type, a
+            // vector with the same guaranteed capacity.
+            // Therefore it is guaranteed that push_back works!
+            static_cast<void>(conditions.push_back(currentCondition));
         }
     }
+
     return conditions;
 }
 
 template <typename WaitFunction>
 WaitSet::ConditionVector WaitSet::waitAndReturnFulfilledConditions(const WaitFunction& wait) noexcept
 {
-    /// @note Inbetween here and last wait someone could have set the trigger to true, hence reset it
-    m_conditionVariableWaiter.reset();
+    WaitSet::ConditionVector conditions;
+
+    if (m_conditionVariableWaiter.wasTriggered())
+    {
+        /// Inbetween here and last wait someone could have set the trigger to true, hence reset it.
+        m_conditionVariableWaiter.reset();
```

Comment:
It seems to be guaranteed by the reset logic that we will not lose wake-ups due to a new condition becoming true, since we check the conditions after reset. It is however hard to reasons whether it is completely behaving as intended (since the intention is not fully clear).